### ISSUE TITLE
Adapter silently stops working and only restarts when disabled and re-enabled again a

### ISSUE THREAD

**Issue by andrenatal** [mozilla-iot/philips-hue-adapter#35]
Every couple of days, this adapter stops working (all the hue lamps things appear as disabled/grayed out and the rules don't work).

The only way to make it back to work is disabling and re-enabling the addon.

**Comment by hobinjk** [comment]
I forgot to update this issue, but this should have been fixed a while ago in 0.7.1. If it's still happening then there's a different root cause from what I expected and I can look into it further.

**Comment by clockbrain** [comment]
I'm seeing similar symptoms although maybe a different cause. Hue adaptor stops working after about 2 hours and needs to be removed and added back in to get it working again.

Here's what I'm seeing in the log file when doing a reinstall of the hue adaptor (v0.7.1).
```
2019-03-26 07:33:01.693 INFO   : Unloading PhilipsHueAdapter
2019-03-26 07:33:01.695 ERROR  : philips-hue: Error updating devices { FetchError: request to http://192.168.0.31:80/api/DyBPYj3wlNrCkWCVXRwByOIkF8dbmV2wMmqe0nj-/lights failed, reason: connect ECONNREFUSED 192.168.0.31:80
2019-03-26 07:33:01.696 ERROR  : philips-hue:     at ClientRequest.<anonymous> (/home/pi/mozilla-iot/gateway/node_modules/node-fetch/lib/index.js:1345:11)
2019-03-26 07:33:01.696 ERROR  : philips-hue:     at emitOne (events.js:116:13)
2019-03-26 07:33:01.697 ERROR  : philips-hue:     at ClientRequest.emit (events.js:211:7)
2019-03-26 07:33:01.697 ERROR  : philips-hue:     at Socket.socketErrorListener (_http_client.js:401:9)
2019-03-26 07:33:01.697 ERROR  : philips-hue:     at emitOne (events.js:116:13)
2019-03-26 07:33:01.698 ERROR  : philips-hue:     at Socket.emit (events.js:211:7)
2019-03-26 07:33:01.698 ERROR  : philips-hue:     at emitErrorNT (internal/streams/destroy.js:66:8)
2019-03-26 07:33:01.699 ERROR  : philips-hue:     at _combinedTickCallback (internal/process/next_tick.js:139:11)
2019-03-26 07:33:01.699 ERROR  : philips-hue:     at process._tickCallback (internal/process/next_tick.js:181:9)
2019-03-26 07:33:01.700 ERROR  : philips-hue:   message: 'request to http://192.168.0.31:80/api/DyBPYj3wlNrCkWCVXRwByOIkF8dbmV2wMmqe0nj-/lights failed, reason: connect ECONNREFUSED 192.168.0.31:80',
2019-03-26 07:33:01.701 ERROR  : philips-hue:   type: 'system',
2019-03-26 07:33:01.701 ERROR  : philips-hue:   errno: 'ECONNREFUSED',
2019-03-26 07:33:01.702 ERROR  : philips-hue:   code: 'ECONNREFUSED' }
2019-03-26 07:33:01.705 INFO   : philips-hue: Adapter: PhilipsHueAdapter unloaded
2019-03-26 07:33:02.704 ERROR  : philips-hue: Error updating devices { FetchError: request to http://192.168.0.31:80/api/DyBPYj3wlNrCkWCVXRwByOIkF8dbmV2wMmqe0nj-/lights failed, reason: connect ECONNREFUSED 192.168.0.31:80
2019-03-26 07:33:02.705 ERROR  : philips-hue:     at ClientRequest.<anonymous> (/home/pi/mozilla-iot/gateway/node_modules/node-fetch/lib/index.js:1345:11)
2019-03-26 07:33:02.705 ERROR  : philips-hue:     at emitOne (events.js:116:13)
2019-03-26 07:33:02.706 ERROR  : philips-hue:     at ClientRequest.emit (events.js:211:7)
2019-03-26 07:33:02.706 ERROR  : philips-hue:     at Socket.socketErrorListener (_http_client.js:401:9)
2019-03-26 07:33:02.707 ERROR  : philips-hue:     at emitOne (events.js:116:13)
2019-03-26 07:33:02.707 ERROR  : philips-hue:     at Socket.emit (events.js:211:7)
2019-03-26 07:33:02.708 ERROR  : philips-hue:     at emitErrorNT (internal/streams/destroy.js:66:8)
2019-03-26 07:33:02.708 ERROR  : philips-hue:     at _combinedTickCallback (internal/process/next_tick.js:139:11)
2019-03-26 07:33:02.708 ERROR  : philips-hue:     at process._tickCallback (internal/process/next_tick.js:181:9)
2019-03-26 07:33:02.709 ERROR  : philips-hue:   message: 'request to http://192.168.0.31:80/api/DyBPYj3wlNrCkWCVXRwByOIkF8dbmV2wMmqe0nj-/lights failed, reason: connect ECONNREFUSED 192.168.0.31:80',
2019-03-26 07:33:02.709 ERROR  : philips-hue:   type: 'system',
2019-03-26 07:33:02.710 ERROR  : philips-hue:   errno: 'ECONNREFUSED',
2019-03-26 07:33:02.710 ERROR  : philips-hue:   code: 'ECONNREFUSED' }
2019-03-26 07:33:32.797 INFO   : Fetching add-on https://github.com/mozilla-iot/philips-hue-adapter/releases/download/v0.7.1/philips-hue-adapter-0.7.1.tgz as /tmp/OmiYGq/philips-hue-adapter.tar.gz
2019-03-26 07:33:36.191 INFO   : Expanding add-on /tmp/OmiYGq/philips-hue-adapter.tar.gz into /home/pi/.mozilla-iot/addons/philips-hue-adapter
2019-03-26 07:33:38.363 INFO   : Loading add-on: philips-hue-adapter
2019-03-26 07:33:39.204 INFO   : philips-hue: Opening database: /home/pi/.mozilla-iot/config/db.sqlite3
2019-03-26 07:33:39.248 INFO   : philips-hue: Loading add-on for philips-hue-adapter from /home/pi/.mozilla-iot/addons/philips-hue-adapter
2019-03-26 07:34:18.455 INFO   : tplink: AddonMangerProxy: recv: b'{"messageType":"setProperty","data":{"deviceId":"tplink-8006F6A2326213CA318A86F391E915451AD1AE59","propertyName":"on","propertyValue":true,"adapterId":"tplink-adapter","pluginId":"tplink-adapter"}}'
2019-03-26 07:34:47.296 INFO   : tplink: AddonMangerProxy: recv: b'{"messageType":"setProperty","data":{"deviceId":"tplink-8006F6A2326213CA318A86F391E915451AD1AE59","propertyName":"on","propertyValue":false,"adapterId":"tplink-adapter","pluginId":"tplink-adapter"}}'
2019-03-26 08:13:14.432 INFO   : getValue for property even_hour for: DateTime returning true
2019-03-26 08:13:14.434 INFO   : getValue for property minute for: DateTime returning 13
2019-03-26 08:13:14.435 INFO   : getValue for property minutes5 for: DateTime returning 10
2019-03-26 08:13:14.437 INFO   : getValue for property hour for: DateTime returning 8
2019-03-26 08:13:15.304 INFO   : getValue for property on for: Back garden light returning false
2019-03-26 08:13:15.306 INFO   : getValue for property level for: Back garden light returning 100
2019-03-26 08:13:15.428 INFO   : getValue for property buttonOn for: Living room switch returning false
2019-03-26 08:13:15.429 INFO   : getValue for property buttonBrighten for: Living room switch returning false
2019-03-26 08:13:15.430 INFO   : getValue for property buttonDim for: Living room switch returning false
2019-03-26 08:13:15.431 INFO   : getValue for property buttonOff for: Living room switch returning false
2019-03-26 08:13:15.433 INFO   : getValue for property battery for: Living room switch returning 100
2019-03-26 08:13:15.680 INFO   : getValue for property on for: Daylight returning false
2019-03-26 08:13:17.082 ERROR  : (node:779) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 actionStatus listeners added. Use emitter.setMaxListeners() to increase limit
2019-03-26 09:42:26.806 INFO   : Checking for add-on updates...
2019-03-26 09:42:27.928 INFO   : Finished updating add-ons
2019-03-26 09:49:03.381 ERROR  : philips-hue: Error updating devices { FetchError: request to http://192.168.0.35:80/api/DyBPYj3wlNrCkWCVXRwByOIkF8dbmV2wMmqe0nj-/sensors failed, reason: connect ECONNREFUSED 192.168.0.35:80
2019-03-26 09:49:03.382 ERROR  : philips-hue:     at ClientRequest.<anonymous> (/home/pi/mozilla-iot/gateway/node_modules/node-fetch/lib/index.js:1345:11)
2019-03-26 09:49:03.383 ERROR  : philips-hue:     at emitOne (events.js:116:13)
2019-03-26 09:49:03.384 ERROR  : philips-hue:     at ClientRequest.emit (events.js:211:7)
2019-03-26 09:49:03.385 ERROR  : philips-hue:     at Socket.socketErrorListener (_http_client.js:401:9)
2019-03-26 09:49:03.386 ERROR  : philips-hue:     at emitOne (events.js:116:13)
2019-03-26 09:49:03.387 ERROR  : philips-hue:     at Socket.emit (events.js:211:7)
2019-03-26 09:49:03.388 ERROR  : philips-hue:     at emitErrorNT (internal/streams/destroy.js:66:8)
2019-03-26 09:49:03.389 ERROR  : philips-hue:     at _combinedTickCallback (internal/process/next_tick.js:139:11)
2019-03-26 09:49:03.390 ERROR  : philips-hue:     at process._tickCallback (internal/process/next_tick.js:181:9)
2019-03-26 09:49:03.391 ERROR  : philips-hue:   message: 'request to http://192.168.0.35:80/api/DyBPYj3wlNrCkWCVXRwByOIkF8dbmV2wMmqe0nj-/sensors failed, reason: connect ECONNREFUSED 192.168.0.35:80',
2019-03-26 09:49:03.392 ERROR  : philips-hue:   type: 'system',
2019-03-26 09:49:03.393 ERROR  : philips-hue:   errno: 'ECONNREFUSED',
2019-03-26 09:49:03.394 ERROR  : philips-hue:   code: 'ECONNREFUSED' }
```
The error then repeats every second spamming the log file.

**Comment by tim-hellhake** [comment]
Same problem here

**Comment by clockbrain** [comment]
Found my problem - the Hue Bridge was getting allocated a different IP address via DHCP so of course the gateway couldn't connect to it. I've fixed the problem by assigning my Hue Bridge a reserved static DHCP address.

@andrenatal don't know if this is the same original issue you were having.

I suggest that the adapter  could handle the error a bit cleaner though. When it can't connect to the Hue Bridge it just keeps trying every second. Here's what it does to my log files. Everything was ok until I enabled the Hue Bridge on the 27th.
```
    4 -rw-r--r--  1 pi pi     2546 Apr 22 19:02 run-app.log.2019-04-22
    4 -rw-r--r--  1 pi pi     1989 Apr 23 19:02 run-app.log.2019-04-23
    4 -rw-r--r--  1 pi pi     2925 Apr 24 19:02 run-app.log.2019-04-24
    4 -rw-r--r--  1 pi pi     1989 Apr 25 19:02 run-app.log.2019-04-25
    4 -rw-r--r--  1 pi pi     1989 Apr 26 19:02 run-app.log.2019-04-26
10248 -rw-r--r--  1 pi pi 10485774 Apr 27 19:14 run-app.log.2019-04-27
10248 -rw-r--r--  1 pi pi 10485819 Apr 27 21:25 run-app.log.2019-04-27.1
10244 -rw-r--r--  1 pi pi 10485789 Apr 27 23:20 run-app.log.2019-04-27.2
 3552 -rw-r--r--  1 pi pi  3634064 Apr 27 23:59 run-app.log.2019-04-27.3
 8960 -rw-r--r--  1 pi pi  9170857 Apr 28 02:02 run-app.log.2019-04-28
```

I'm thinking somewhere in this code [https://github.com/mozilla-iot/philips-hue-adapter/blob/c17bf90093de19a8957ad0de86b6c98f950ced41/philips-hue-adapter.js#L725](https://github.com/mozilla-iot/philips-hue-adapter/blob/c17bf90093de19a8957ad0de86b6c98f950ced41/philips-hue-adapter.js#L725) it should maybe increase the update interval if it can't connect?

**Comment by mrstegeman** [comment]
Fixed in c5ab17a2f9bfb4521e66bb446ed8a1aa3d4a477d
